src/core/NewRestriction.ml

```diff
@@ -154,13 +154,13 @@ let equate' x y h =
   get_m h @@ equate x y h
 
 let subst (r : dim) (x : atom) (h : t) =
-  if mem_atom x h.index then
+  if mem_atom x h.index && r <> `Atom x then
```

Comment:
@jonsterling I was blessed by the God of Syntax today!

Reply:
:cloud_with_lightning: :crown: